stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -87,10 +87,195 @@ public enum _ExecutionMode : Equatable {
   }
 }
 
+// TraceContext contains the state as needed to build a trace graph function
+// (TF_Function). As eager ops are executed in tracing mode, their corresponding
+// nodes are added to the trace graph (via `addEagerOpToGraph()`). When the
+// trace is fully formed (via `finalizeAndExecute()`), the trace graph function
+// is run by the eager runtime.
+private class TraceContext {
+  let status: CTFStatus = TF_NewStatus()
+
+  // The trace graph, which will be converted to a trace graph function
+  // (TF_Function) upon finalizing.
+  let graph = TF_NewGraph()
+
+  // Used to create unique trace graph node names and graph function names.
+  var traceGraphObjectCounter = 0
```

Comment:
I don't see this used to uniquify node names. Would you not need this entirely if you used append_hash_to_fn_name below?

Reply:
I moved this to `_RuntimeConfig` so that it lives across the lifetimes of `TraceContext` instances. Also renamed it to `traceGraphFunctionCounter`. 

We don't need a "trace graph node counter" at this swift level (any more), since that's done at the C API impl of `TFE_AddEagerOpToGraph()`.